watchtower/wtmock/signer.go

```diff
@@ -47,6 +48,69 @@ func (s *MockSigner) SignOutputRaw(tx *wire.MsgTx,
 		panic("cannot sign w/ unknown key")
 	}
 
+	// In case of a taproot output any signature is always a Schnorr
+	// signature, based on the new tapscript sighash algorithm.
+	if txscript.IsPayToTaproot(signDesc.Output.PkScript) {
+		sigHashes := txscript.NewTxSigHashes(
```

Comment:
We've copied this over in a few places in the codebase, perhaps we can try to unify with an `internal/mock` package?

Reply:
Ahh true re import cycle, I think I ran into the same thing when adding musig2 awareness in the first place...

Perhaps we can just make a new issue here to attempt to conslidate the mock signers in a single place. We don't update it all that often, but it can def burn some dev time when something small changes, and one mock signer works, but not the other.